src/application/components/Layouts/SettingsModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { Dispatch, Fragment, MutableRefObject, SetStateAction } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+declare const VERSION: string;
+
+export function Modal({
+  open,
+  setOpen,
+  cancelButtonRef,
+}: {
+  open: boolean;
+  cancelButtonRef: MutableRefObject<null>;
+  setOpen: Dispatch<SetStateAction<boolean>>;
```

Comment:
[nit] I tend to like standardizing on event callbacks from child -> parent with the `on<Event>` naming scheme to be able to pick out which are the event callbacks. I also prefer having the function signature a bit more generic so you aren't coupling it specifically to `useState`. It can be refactored much easier this way if we ever need to (e.g. if you needed to execute a side effect when closing the modal). 

```suggestion
  onOpen: (open: boolean) => void
```

`Dispatch<SetStateAction<boolean>>` should play just fine with this function signature, so passing `setOpen` directly to `onOpen` from the parent works.

```tsx
<Modal onOpen={setOpen} /> // works!
```